### PR TITLE
Add upload of files

### PR DESCRIPTION
**Pull request by jekuaitk**
* Adds `AddToDocumentLibrary` method to `Documents` service

### CODE REVIEW

File: src/Service/Documents.php [hunk 1]
```diff
@@ -38,4 +39,63 @@ public function Metadata(int $docId)
...
     }
+
+    /**
+     * @throws GetOrganizedClientException
+     * @throws InvalidFilePathException
+     */
+    public function AddToDocumentLibrary(string $filePath, string $caseId, string $fileName = '', array $metaDataOptions = [], bool $overWrite = true, string $listName = 'Dokumenter', string $folderPath = '')
+    {
+        if (!file_exists($filePath)) {
+            throw new InvalidFilePathException(sprintf('File: %s does not exist', $filePath));
```
Comment (rimi-itk) [inline src/Service/Documents.php]: Skip the colon.

File: src/Service/Documents.php [hunk 2]
```diff
@@ -38,4 +39,63 @@ public function Metadata(int $docId)
...
+
+        // File does exist, transform it into byte array via stream
+        $fileByteArray = [];
+        $stream = fopen($filePath, 'r');
+
+        while (!feof($stream)) {
+            $fileByteArray = array_merge($fileByteArray, $this->stringToByteArray(fgets($stream)));
+        }
+
+        fclose($stream);
```
Comment (rimi-itk) [inline src/Service/Documents.php]: Move all this into a single function. Maybe something like https://github.com/itk-dev/getorganized-api-client-php/blob/3c8bbd788a4e7a6bb9c0b095ee6a207d64b9976a/src/Mock/RequestHelper.php#L44-L60.

For performance reasons you should be careful with calling `array_merge` inside a loop.

File: src/Service/Documents.php [hunk 3]
```diff
@@ -38,4 +39,63 @@ public function Metadata(int $docId)
 
         throw new InvalidResponseException('Metadata missing in response');
     }
+
+    /**
+     * @throws GetOrganizedClientException
+     * @throws InvalidFilePathException
+     */
+    public function AddToDocumentLibrary(string $filePath, string $caseId, string $fileName = '', array $metaDataOptions = [], bool $overWrite = true, string $listName = 'Dokumenter', string $folderPath = '')
```
Comment (rimi-itk) [inline src/Service/Documents.php]: I suggest renaming `$metadataOptions` to just `$metadata`. And `$overWrite` to `$overwrite`.

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: src/Service/Documents.php (hunk 3 above)
Comment (jekuaitk) [reply src/Service/Documents.php]: Renamed.

File: src/Service/Documents.php (hunk 1 above)
Comment (jekuaitk) [reply src/Service/Documents.php]: Skipped!

File: src/Service/Documents.php (hunk 2 above)
Comment (jekuaitk) [reply src/Service/Documents.php]: Cleaned up code as proposed.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**